src/v/kafka/server/handlers/configs/config_utils.h

```diff
@@ -473,6 +474,27 @@ struct iceberg_config_validator {
     }
 };
 
+struct leaders_preference_validator {
```

Comment:
We want to use this validator on topic creation as well, in addition to IncrementalAlter+Alter, right?

Reply:
my understanding is that in sanctioned mode you are not allowed to create topics, at all.
@BenPope is that correct, or am i misunderstanding [CORE-7999](https://redpandadata.atlassian.net/browse/CORE-7999)

Reply:
As the pin leadership part has been moved out of this pr, this piece of work will get transferred there.